sdk/eventhubs/azure_messaging_eventhubs/test-resources.bicep

```diff
@@ -0,0 +1,236 @@
+  name: guid(eventhubNamespace.id, 'Storage Blob Data Owner')
+
+  properties: {
+    roleDefinitionId: subscriptionResourceId(
+      'Microsoft.Authorization/roleDefinitions',
+      'b7e6dc6d-f1e8-4753-8033-0f276bb0955b'
+    ) // Storage Blob Data Owner
+    principalType: 'User'
+    principalId: testApplicationOid
+  }
+  dependsOn: [
+    eventHub
+    storageAccount
+  ]
+}
+
+resource eventHub 'Microsoft.EventHub/namespaces/eventhubs@2024-05-01-preview' = {
+  parent: eventhubNamespace
+  name: eventHubName
+  properties: {
+    messageTimestampDescription: {
+      timestampType: 'LogAppend'
+    }
+    retentionDescription: {
+      cleanupPolicy: 'Delete'
+      retentionTimeInHours: 24
+    }
+    messageRetentionInDays: 1
+    partitionCount: 4
+    status: 'Active'
+  }
+}
+
+resource eventhubNamespace_networkruleset_default 'Microsoft.EventHub/namespaces/networkrulesets@2024-05-01-preview' = {
+  parent: eventhubNamespace
+  name: 'default'
+  properties: {
+    publicNetworkAccess: 'Enabled'
+    defaultAction: 'Allow'
+    virtualNetworkRules: []
+    ipRules: []
+    trustedServiceAccessEnabled: false
+  }
+}
+
+resource consumerGroup 'Microsoft.EventHub/namespaces/eventhubs/consumergroups@2024-05-01-preview' = {
+  parent: eventHub
+  name: '$Default'
+  properties: {}
+}
+
+resource defaultGroup 'Microsoft.EventHub/namespaces/eventhubs/consumergroups@2024-05-01-preview' = {
+  parent: eventHub
+  name: 'defaultgroup'
+  properties: {}
+}
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2024-01-01' = {
+  name: storageAccountName
+  location: location
+  sku: {
+    name: 'Standard_LRS'
+  }
+  kind: 'BlobStorage'
+  properties: {
+    allowCrossTenantReplication: false
+    minimumTlsVersion: 'TLS1_2'
+    allowBlobPublicAccess: false
+    allowSharedKeyAccess: false
+    networkAcls: {
+      bypass: 'AzureServices'
+      virtualNetworkRules: []
+      ipRules: []
+      defaultAction: 'Allow'
+    }
+    supportsHttpsTrafficOnly: true
+    encryption: {
+      services: {
+        file: {
+          keyType: 'Account'
+          enabled: true
+        }
+        blob: {
+          keyType: 'Account'
+          enabled: true
+        }
+      }
+      keySource: 'Microsoft.Storage'
+    }
+    accessTier: 'Hot'
+  }
+}
+
+resource storageAccount_default 'Microsoft.Storage/storageAccounts/blobServices@2024-01-01' = {
+  parent: storageAccount
+  name: 'default'
+  properties: {
+    cors: {
+      corsRules: []
+    }
+    deleteRetentionPolicy: {
+      allowPermanentDelete: false
+      enabled: false
+    }
+  }
+}
+
+resource storageAccount_blobContainer 'Microsoft.Storage/storageAccounts/blobServices/containers@2019-04-01' = {
+  name: 'container'
+  parent: storageAccount_default
+  properties: {}
+}
+
+// Outputs
+output EVENTHUB_NAME string = eventHub.name
+output EVENTHUBS_NAMESPACE string = eventhubNamespace.name
+output EVENTHUBS_HOST string = '${eventhubNamespace.name}.servicebus.windows.net'
```

Comment:
The full URI isn't available in case we provision in another cloud?

Reply:
There is a "serviceBusEndpoint" URL but it's a URL in the HTTPS namespace. And I don't think I can extract just the host from a string in bicep.